.github/workflows/pypi.yaml

```diff
@@ -0,0 +1,127 @@
+          manylinux: auto
+          maturin-version: latest
+          # Maturin's default command is 'build',
+          # and target supported python version ('-i' flag) and store in dist, so we can upload all wheels at same time.
+          args: --release -i ${{ matrix.py-version }} --out dist
+      - name: Upload wheels
+        uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: dist
+
+
+  # Windows job is pretty straight forward, no containers. Runs on Windows, for Windows.
+  windows:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        target: [ x64, x86 ]
+        py-version: [ '3.7', '3.8', '3.9', '3.10' ]
+    steps:
+      - uses: actions/checkout@v2
+        # since we don't have a requirements.txt file in the project root, setup-python needs one.
+      - name: check requirements.txt
+        run: |
+          if (!(Test-Path requirements.txt))
+          {
+          New-Item -itemType File -Name requirements.txt
+          }
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+          architecture: ${{ matrix.target }}
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          default: true
+      - name: Build wheels
+        uses: messense/maturin-action@v1
+        with:
+          target: ${{ matrix.target }}
+          maturin-version: latest
+          args: --release -i ${{ matrix.py-version }} --out dist
+      - name: Upload wheels
+        uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: dist
+
+  macos:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+        # since we don't have a requirements.txt file in the project root, setup-python needs one.
+      - name: check requirements.txt
+        run: |
+          if [[ ! -f requirements.txt ]]; then
+            touch requirements.txt
+          fi
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install Rust toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          default: true
+      - name: Build wheels - universal2
+        uses: messense/maturin-action@v1
+        with:
+          args: --release --universal2 --out dist
+      - name: Upload wheels
+        uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: dist
+
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: [ macos, linux, windows ]
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: wheels
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Publish to PyPI
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```

Comment:
I'm wondering if there's a way to use some kind of API key. tbh this makes me want to set us up with an org and just marry it to that. i think i can bite the bullet for now.

Reply:
Hmm you mention an API token for the project. Is `__token__ supposed to be that?

> TWINE_USERNAME

Reply:
I need to read more thoroughly. The PW is the token. interesting how it's set up as username and pw

Reply:
Honestly i think you had it right. i was just reading the pypi documentation about it.

> To use this API token:
>
>    Set your username to \_\_token\_\_
>    Set your password to the token value, including the pypi- prefix